### PR TITLE
Jhakulin/parse messages

### PR DESCRIPTION
**Pull request by jhakulin**
1. Added following agent operation methods:

def get_file_content_stream(self, file_id: str, **kwargs: Any) -> Iterator[bytes]:
  - This will enable file content download using streaming. Currently streaming is required for file content download.
 
def save_file(
        self, 
        file_id: str, 
        file_name: str, 
        target_dir: Optional[Union[str, Path]] = None
    ) -> None:
  - Saves file content retrieved using a file identifier to the specified local directory.

def get_messages(
        self,
        thread_id: str,
        *,
        run_id: Optional[str] = None,
        limit: Optional[int] = None,
        order: Optional[Union[str, _models.ListSortOrder]] = None,
        after: Optional[str] = None,
        before: Optional[str] = None,
        **kwargs: Any
    ) -> _models.Messages:
        - Parses the OpenAIPageableListOfThreadMessage response and returns a Messages object. 

2. Added Messages class that represents a collection of messages from agent and parses them into convenient properties.
3. Updated CodeInterpreter sample to produce bar chart based on given csv file

### CODE REVIEW

File: sdk/ai/azure-ai-projects/azure/ai/projects/models/_patch.py [hunk 1]
```diff
@@ -991,6 +997,94 @@ def until_done(self) -> None:
             pass
 
 
+class Messages:
```
Comment (dargilco) [inline sdk/ai/azure-ai-projects/azure/ai/projects/models/_patch.py:1000]: One of the frequent comments in SDK reviews is about one-work class names, that are very generic.  Can we make this a two-word class name to be more specific?

Comment (jhakulin) [reply sdk/ai/azure-ai-projects/azure/ai/projects/models/_patch.py:1000]: I try to get Johan comments, also will think about more specific name

**Review by nick863 — CHANGES_REQUESTED** [review]

File: sdk/ai/azure-ai-projects/azure/ai/projects/models/_patch.py [hunk 2]
```diff
@@ -419,8 +425,8 @@ class CodeInterpreterTool(Tool):
     A tool that interprets code files uploaded to the agent.
     """
 
-    def __init__(self):
-        self.file_ids = []
+    def __init__(self, file_ids: List[str] = []):
```
[the remark this answers is not in the document]
Comment (jhakulin) [reply sdk/ai/azure-ai-projects/azure/ai/projects/models/_patch.py:428]: oh, you mean we shall add check in add method to not add if the file_id already exists in the list?
That would be good.

File: sdk/ai/azure-ai-projects/azure/ai/projects/models/_patch.py [hunk 3]
```diff
@@ -991,6 +997,94 @@ def until_done(self) -> None:
...
+    def get_last_text_message_by_sender(self, sender: str) -> Optional[MessageTextContent]:
+        """Returns the last text message from the specified sender.
+        
+        :param sender: The role of the sender.
+        :type sender: str
+
+        :return: The last text message from the specified sender.
+        :rtype: ~azure.ai.projects.models.MessageTextContent
+        """
+        for msg in (self._messages):
```
[the remark this answers is not in the document]
Comment (jhakulin) [reply sdk/ai/azure-ai-projects/azure/ai/projects/models/_patch.py:1100]: agent returns already reversed list

File: sdk/ai/azure-ai-projects/azure/ai/projects/operations/_patch.py [hunk 4]
```diff
@@ -2013,6 +2017,117 @@ def create_vector_store_file_batch_and_poll(
...
+        :raises ~azure.core.exceptions.HttpResponseError: If the HTTP request fails.
+        """
+        kwargs['stream'] = True
+        
+        try:
+            response = super().get_file_content(file_id, **kwargs)
+            stream_iterator: Iterator[bytes] = cast(Iterator[bytes], response)
+            return stream_iterator
+        except HttpResponseError as e:
+            raise e
```
[the remark this answers is not in the document]
Comment (jhakulin) [reply sdk/ai/azure-ai-projects/azure/ai/projects/operations/_patch.py]: fixed

File: sdk/ai/azure-ai-projects/azure/ai/projects/operations/_patch.py [hunk 5]
```diff
@@ -2013,6 +2017,117 @@ def create_vector_store_file_batch_and_poll(
...
+        :return: An iterator that yields bytes from the file content.
+        :rtype: Iterator[bytes]
+        :raises ~azure.core.exceptions.HttpResponseError: If the HTTP request fails.
+        """
+        kwargs['stream'] = True
+        
+        try:
+            response = super().get_file_content(file_id, **kwargs)
+            stream_iterator: Iterator[bytes] = cast(Iterator[bytes], response)
+            return stream_iterator
```
[the remark this answers is not in the document]
Comment (jhakulin) [reply sdk/ai/azure-ai-projects/azure/ai/projects/operations/_patch.py]: fixed

File: sdk/ai/azure-ai-projects/azure/ai/projects/models/_patch.py (hunk 1 above)
Comment (jhakulin) [reply sdk/ai/azure-ai-projects/azure/ai/projects/models/_patch.py]: Changed to ThreadMessages

File: sdk/ai/azure-ai-projects/azure/ai/projects/models/_patch.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jhakulin) [reply sdk/ai/azure-ai-projects/azure/ai/projects/models/_patch.py]: Fixed

File: sdk/ai/azure-ai-projects/azure/ai/projects/operations/_patch.py [hunk 6]
```diff
@@ -2013,6 +2017,117 @@ def create_vector_store_file_batch_and_poll(
...
+        thread_id: str,
+        *,
+        run_id: Optional[str] = None,
+        limit: Optional[int] = None,
+        order: Optional[Union[str, _models.ListSortOrder]] = None,
+        after: Optional[str] = None,
+        before: Optional[str] = None,
+        **kwargs: Any
+    ) -> _models.Messages:
+        """Parses the OpenAIPageableListOfThreadMessage response and returns a Messages object.
```
[the remark this answers is not in the document]
Comment (jhakulin) [reply sdk/ai/azure-ai-projects/azure/ai/projects/operations/_patch.py]: fixed, thank you!

**Review by dargilco — APPROVED** [review]

**Review by nick863 — APPROVED** [review]

**Pull request merged**